test-suite/output/PrintMatch.out

```diff
@@ -22,3 +22,27 @@ end
        P a (srefl a) -> forall (a0 : A) (s : seq a a0), P a0 s
 
 Arguments seq_rect A%type_scope a P%function_scope f a0 s
+eq_sym =
+fun (A : Type) (x y : A) (H : @eq A x y) =>
+match H in (@eq _ _ a) return (@eq A a x) with
+| @eq_refl _ _ => @eq_refl A x
+end
+     : forall (A : Type) (x y : A) (_ : @eq A x y), @eq A y x
+
+Arguments eq_sym [A]%type_scope [x y] _
+eq_sym =
+fun (A : Type) (x y : A) (H : x = y) =>
+match H in (_ = a) return (a = x) with
+| @eq_refl _ _ => @eq_refl A x
+end
+     : forall (A : Type) (x y : A), x = y -> y = x
+
+Arguments eq_sym [A]%type_scope [x y] _
+eq_sym =
+fun (A : Type) (x y : A) (H : x = y) =>
+match H in (_ = a) return (a = x) with
+| @eq_refl _ _ => @eq_refl A x
```

Comment:
Is this correct with asymmetric patterns?

Reply:
I was surprised but apparently yes. This was a change between Coq 8.4 to 8.5 (at the time of the asymmetric change).

8.4: 
- `Check fun l:list nat => match l with nil => nil | @cons _ _ _ => @nil nat end.`  fails
- `Check fun l:list nat => match l with nil => nil | @cons _ _ => @nil nat end.`  works

8.5:
- `Check fun l:list nat => match l with nil => nil | @cons _ _ _ => @nil nat end.`  works
- `Check fun l:list nat => match l with nil => nil | @cons _ _ => @nil nat end.`  fails